crates/cairo-lint-core/src/lints/double_comparison.rs

```diff
@@ -0,0 +1,38 @@
+use cairo_lang_defs::plugin::PluginDiagnostic;
+use cairo_lang_diagnostics::Severity;
+use cairo_lang_syntax::node::ast::{BinaryOperator, Expr};
+use cairo_lang_syntax::node::db::SyntaxGroup;
+use cairo_lang_syntax::node::{TypedStablePtr, TypedSyntaxNode};
+
+pub const DOUBLE_COMPARISON: &str = "redundant double comparison found. Consider simplifying to a single comparison.";
+
+pub fn check_double_comparison(db: &dyn SyntaxGroup, expr: &Expr, diagnostics: &mut Vec<PluginDiagnostic>) {
+    if let Expr::Binary(binary_op) = expr {
```

Comment:
you need to check that the 2 variables compared are the same for the 2 comparisons